File: 1064-smallest-integer-divisible-by-k/smallest-integer-divisible-by-k.py
from itertools import accumulate as acc

class Solution:
    def smallestRepunitDivByK(self, k: int) -> int:
        return min(
            (
                i + 1
                for i, r in enumerate(
                    acc(range(1, k + 3), lambda a, b: (a * 10 + 1) % k)
                )
                if r % k == 0
            ),
            default=-1,
        )

File: 1064-smallest-integer-divisible-by-k/smallest-integer-divisible-by-k.go
func smallestRepunitDivByK(k int) int {
	if 1%k == 0 {
		return 1
	}

	for n, i := 1, 1; i <= k; n, i = ((n*10 + 1) % k), i+1 {
		if n == 0 {
			return i
		}
	}
    
	return -1
}